### PR TITLE
Day01 update

### PR DESCRIPTION
**Pull request by ssingal05**

### CODE REVIEW

File: day01/README.md [hunk 1]
```diff
@@ -1,8 +1,71 @@
-# Day 01: Implementing an Arraylist
+# `ArrayList`s
 
-## MyArrayList
-* get: implement the `get` method of `MyArrayList`. Refer to the Java API to figure out what this method does.
-* indexOf: implement the `indexOf` method of `MyArrayList`.
-* add: implement the `add` method of `MyArrayList`.
-* remove: implement the `remove` method of `MyArrayList`.
+In this homework assignment, you are going to given the class definition and some method definitions of `MyArrayList`
```
Comment (thecardkid) [inline day01/README.md]: Change to "you are going to write your own ArrayList class"

File: day01/README.md [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
+public class MyArrayList {
+...
+}
+```
+
+With each problem, we will add functionality to our classes to more closely match the real functionality of a real Java `ArrayList`.
+
+### `MyArrayList`
+
+Joe would like to start a farm that only contains cows. He wants your help in creating an `ArrayList` to store these cows in.
```
Comment (thecardkid) [inline day01/README.md]: Just a suggestion: what if you change "Joe" to "Sidd" ?

File: day01/solutions/MyArrayList.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
+public class MyArrayList<T> {
+
+	private Object[] elems;
+	private int numElems;
+
+	public MyArrayList(){
+
+		elems = new Object[10];
```
Comment (thecardkid) [inline day01/solutions/MyArrayList.java]: Can you make a note to cover array literals in class

File: day01/solutions/MyArrayList.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	public int size(){
+
+		return numElems;
+
+	}
+
+	public T get(int index){
+
+		if(index >= numElems) throw new IndexOutOfBoundsException("Index out of bounds.");
+		return (T)elems[index];
```
Comment (thecardkid) [inline day01/solutions/MyArrayList.java]: Cover casting in class. I think you should explicitly say something like "the AL class can only wrap Object arrays"

File: day01/README.md (hunk 2 above)
Comment (davidabrahams) [inline day01/README.md]: hahahahaha i approve of this change

**Pull request merged**